src/test/java/org/broadinstitute/hellbender/tools/reference/ReferenceSequenceTableUnitTest.java

```diff
@@ -0,0 +1,228 @@
+package org.broadinstitute.hellbender.tools.reference;
+
+import org.broadinstitute.hellbender.engine.GATKPath;
+import org.broadinstitute.hellbender.engine.ReferenceDataSource;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+public class ReferenceSequenceTableUnitTest {
+
+    public ReferenceSequenceTable tableGenerator(List<GATKPath> references, CompareReferences.MD5CalculationMode md5CalculationMode){
+        Map<GATKPath, ReferenceDataSource> referenceSources = new LinkedHashMap<>();
+        for(GATKPath path : references){
+            referenceSources.put(path, ReferenceDataSource.of(path.toPath()));
+        }
+
+        ReferenceSequenceTable table = new ReferenceSequenceTable(referenceSources, md5CalculationMode);
+        table.build();
+
+        return table;
+    }
+
+    @DataProvider(name = "testQueryByMD5Data")
+    public Object[][] testQueryByMD5Data() {
+        return new Object[][]{
+                // references, md5, length, table
+                new Object[]{ "8c0c38e352d8f3309eabe4845456f274",
+                        16000,
+                        tableGenerator(Arrays.asList(new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini.fasta")),
+                                CompareReferences.MD5CalculationMode.USE_DICT),
+                },
+                new Object[]{ "8c0c38e352d8f3309eabe4845456f274",
+                        16000,
+                        tableGenerator(Arrays.asList(new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini.fasta"),
+                                new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini_1renamed.fasta")),
+                                CompareReferences.MD5CalculationMode.USE_DICT),
+                },
+        };
+    }
+
+    public boolean compareRows(ReferenceSequenceTable.TableRow row1, ReferenceSequenceTable.TableRow row2){
+        return row1.getMd5().equals(row2.getMd5()) && row1.getLength() == row2.getLength();
```

Comment:
Add a comment documenting why we can't use the TableRow equals() method here

Reply:
@orlicohen